File: src_all/sources/MobileIMSDKServer/src/net/openmob/mobileimsdk/server/protocal/s/PErrorResponse.java
/*
 * Copyright (C) 2016 即时通讯网(52im.net) The MobileIMSDK Project. 
 * All rights reserved.
 * Project URL:https://github.com/JackJiang2011/MobileIMSDK
 *  
 * 即时通讯网(52im.net) - 即时通讯技术社区! PROPRIETARY/CONFIDENTIAL.
 * Use is subject to license terms.
 * 
 * PErrorResponse.java at 2016-2-20 11:26:02, code by Jack Jiang.
 * You can contact author with jack.jiang@52im.net or jb2011@163.com.
 */
package net.openmob.mobileimsdk.server.protocal.s;

public class PErrorResponse
{
	private int errorCode = -1;
	private String errorMsg = null;

	public PErrorResponse(int errorCode, String errorMsg)
	{
		this.errorCode = errorCode;
		this.errorMsg = errorMsg;
	}

	public int getErrorCode()
	{
		return this.errorCode;
	}

	public void setErrorCode(int errorCode)
	{
		this.errorCode = errorCode;
	}

	public String getErrorMsg()
	{
		return this.errorMsg;
	}

	public void setErrorMsg(String errorMsg)
	{
		this.errorMsg = errorMsg;
	}
}

File: sdk_src/TCP_Client/MobileIMSDK4f_tcp_Open(Flutter版,由社区网友开发,仅作学习参考)/lib/server/protocol/s/p_error_response.dart
import 'dart:core';

class PErrorResponse {
  int _errorCode = -1;
  String? _errorMsg;

  PErrorResponse(this._errorCode,this._errorMsg);

  //命名式构造方法，也可以是用工厂构造方法
  PErrorResponse.fromJson(Map<String, dynamic> json)
      : _errorCode = json['errorCode'],
        _errorMsg = json['errorMsg'];

  //如果想写成协议，归档(json和对象互转时，为了使用方便)时，可以继承协议，那么可以用普通方法，而不是构造方法
  Map<String, dynamic> toJson() => {
    'errorCode': _errorCode,
    'errorMsg': _errorMsg
  };

  getErrorCode() {
    return _errorCode;
  }

  void setErrorCode(int errorCode) {
    _errorCode = errorCode;
  }

  getErrorMsg() {
    return _errorMsg;
  }

  void setErrorMsg(String errorMsg) {
    _errorMsg = errorMsg;
  }
}
